test/testother.cpp

```diff
@@ -4374,6 +4375,17 @@ class TestOther : public TestFixture {
         ASSERT_EQUALS("", errout.str());
     }
 
+    void suspiciousUnaryPlusMinus() { // #8004
+        check("int g() { return 1; }\n"
+              "void f() {\n"
+              "    +g();\n"
+              "    -g();\n"
+              "}\n");
+        ASSERT_EQUALS("[test.cpp:3]: (warning, inconclusive) Found suspicious operator '+'\n"
```

Comment:
not your code.. but the warning message should be more clear. Would it be possible to write "result is not used" also?